vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/machinehealthcheck/machinehealthcheck.go

```diff
@@ -0,0 +1,217 @@
+package machinehealthcheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/glog"
+	e2e "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	healthcheckingv1alpha1 "github.com/openshift/machine-api-operator/pkg/apis/healthchecking/v1alpha1"
+	"github.com/openshift/machine-api-operator/pkg/util/conditions"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck controller", func() {
+	var client runtimeclient.Client
+	var numberOfReadyWorkers int
+	var workerNode *corev1.Node
+	var workerMachine *mapiv1beta1.Machine
+
+	waitNodeUnhealthyCondition := func() {
+		key := types.NamespacedName{
+			Name:      workerNode.Name,
```

Comment:
Can you make this (`workerNode.Name`) an argument of waitNodeUnhealthyCondition function? So there are no "magic" values read outside of well-defined function signature. It's even fine to move `waitNodeUnhealthyCondition` out of `Describe`.